crates/ruff_python_formatter/src/other/parameters.rs

```diff
@@ -676,3 +677,10 @@ fn has_trailing_comma(
         .kind()
         == SimpleTokenKind::Comma
 }
+
+/// Returns `true` if the parameters are parenthesized (as in a function definition), or `false` if
+/// not (as in a lambda).
+pub(crate) fn are_parameters_parenthesized(parameters: &Parameters, locator: &Locator) -> bool {
+    // A lambda never has parentheses around its parameters, but a function definition always does.
+    locator.slice(parameters.range).starts_with('(')
```

Comment:
Nit: Unrelated to your PR. It would be nice if `slice` accepts `Into<Ranged>` but that probably requires moving `Ranged` to `ruff_text_size`

Reply:
I might do this, I think `Ranged` _should_ be in `ruff_text_size`.

Reply:
@zanieb probably gonna do this per offline conversation.